pkg/tools/interfaces.go

```diff
@@ -46,3 +47,11 @@ type EtcdClient interface {
 	// the etcd client interface which doesn't, and it doesn't seem worth it to wrap the api.
 	Watch(prefix string, waitIndex uint64, recursive bool, receiver chan *etcd.Response, stop chan bool) (*etcd.Response, error)
 }
+
+func NewLocalTestEtcdClient(t *testing.T) EtcdClient {
```

Comment:
I think we should move this to:
pkg/storage/etcd/testing
or sth like that

I would like to avoid mixing production and test code in the same files.